[Python] Overhaul "Setup tracing" > "Custom instrumentation" (#11341)
getsentry/sentry-docs: docs/platforms/python/tracing/instrumentation/custom-instrumentation/index.mdx

    with sentry_sdk.start_transaction(op="task", name="Eat Pizza"):
        while pizza.slices > 0:
            eat_slice(pizza.slices.pop())
```

<Alert title="Static & Class Methods" level="info">

When tracing a static or class method, you **must** add the `@sentry_sdk.trace` decorator **after** the `@staticmethod` or `@classmethod` decorator (i.e., **closer** to the function definition). Otherwise, your function will break!

</Alert>

> ### Manually

Reviewer: I think this heading is a bit confusing because the other methods described on this page are also different forms of "manual" instrumentation. Maybe it would be clearer to call this "Manually starting and stopping the span" or something similar

Author: done

```python
import sentry_sdk

def eat_slice(slice):
    ...

def eat_pizza(pizza):
    with sentry_sdk.start_transaction(op="task", name="Eat Pizza"):
        while pizza.slices > 0:

>             span = sentry_sdk.start_span(description="Eat Slice")

Reviewer: Is it necessary to call `__enter__` and `__exit__` here? We set some things on the scope in those magic methods, is that important?

            eat_slice(pizza.slices.pop())
            span.finish()
```

When you create your span manually, make sure to call `span.finish()` after the block of code you want to wrap in a span to finish the span. If you do not finish the span it will not be sent to Sentry.

## Nested Spans

> Spans can be nested to form a span tree. If you'd like to learn more, read our [distributed tracing](/product/sentry-basics/tracing/distributed-tracing/) documentation.

Reviewer: I don't think the examples in this section are needed, we could probably delete them

Now, whenever a function specified in `functions_to_trace` will be executed, a span will be created and attached as a child to the currently running span.

> <Alert title="Important" level="info">
> 
> To enable performance monitoring for the functions specified in `functions_to_trace`, the SDK needs to load the function modules. Be aware, there may be code being executed in modules during module loading. To avoid this, use the method described above to trace your functions.
> 
> </Alert>

Reviewer: I feel like this page already has so much information. This block will just confuse people. We should either remove it, or if it is important, let's describe `functions_to_trace` in depth on a different page.

```suggestion
```

> ## Accessing the Current Transaction

Reviewer: I think we could probably also delete this section or move it to its own page. This is probably too much information for most use cases, especially because users who are doing custom instrumentation should already have a reference to the current transaction, since `start_transaction` returns the transaction.

> To change data in an already ongoing transaction, use `sentry_sdk.get_current_scope().transaction`. This property will return a transaction if there's one running, otherwise it will return `None`.

Reviewer: ```suggestion
The `sentry_sdk.get_current_scope().transaction` property returns the active transaction or `None` if no transaction is active. You can use this property to modify data on the transaction.
```

> ## Accessing the Current Span

Reviewer: Same here, this section is again possibly too detailed for this page.

> To change data in the current span, use ` sentry_sdk.get_current_scope().span`. This property will return a span if there's one running, otherwise it will return `None`.

Reviewer 2: ```suggestion
To change data in the current span, use ` sentry_sdk.get_current_span()`. This function will return a span if there's one running, otherwise it will return `None`.
```

@sentry_sdk.trace
def eat_slice(slice):

>     span = sentry_sdk.get_current_scope().span

Reviewer 2: ```suggestion
    span = sentry_sdk.get_current_span()
```
